### PR TITLE
MAINT: pytest

### PR DESCRIPTION
**Pull request by wasade**
* Support for legacy numpy repr strings
* Shifting to pytest
* Update doctest strings to support py3
* Added in py37 into the build matrix

We are still seeing a lot of warnings surrounding the matrix use which I *believe* are stemming from scipy.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -107,16 +142,17 @@
       maintainer=__maintainer__,
       maintainer_email=__email__,
       url='http://www.biom-format.org',
-      test_suite='nose.collector',
       packages=find_packages(),
+      tests_require=['pytest', 'pytest-cov'],
```
Comment (ElDeveloper) [inline setup.py]: can you add the missing test's dependencies in this list i.e. nose and flake8?

File: setup.py [hunk 2]
```diff
@@ -107,16 +142,17 @@
...
       include_package_data=True,
       ext_modules=extensions,
       include_dirs=[np.get_include()],
       install_requires=install_requires,
-      extras_require={'test': ["nose >= 0.10.1", "flake8"],
+      extras_require={'test': ["flake8"],
```
Comment (ElDeveloper) [inline setup.py]: Remove this.

File: .travis.yml [hunk 3]
```diff
@@ -1,39 +1,34 @@
...
 language: python
 env:
-  - PYTHON_VERSION=2.7 USE_H5PY=True NOSE_ARGS="--with-doctest --with-coverage"
-  - PYTHON_VERSION=2.7 USE_CYTHON=True NOSE_ARGS="--with-doctest --with-coverage"
-  - PYTHON_VERSION=3.5 USE_H5PY=True
-  - PYTHON_VERSION=3.5 USE_CYTHON=True
-  - PYTHON_VERSION=3.6 USE_H5PY=True
-  - PYTHON_VERSION=3.6 USE_CYTHON=True
+  - PYTHON_VERSION=2.7 USE_H5PY=True USE_CYTHON=True WITH_DOCTEST=False
+  - PYTHON_VERSION=3.5 USE_H5PY=True USE_CYTHON=True WITH_DOCTEST=True
```
[the remark this answers is not in the document]
Comment (wasade) [reply .travis.yml]: Done

File: ChangeLog.md [hunk 4]
```diff
@@ -8,7 +8,8 @@ Changes since 2.1.6 go here.
 
 Important: 
 
-* Python 3.4 support has been dropped. We now only support Python 2.7, 3.5 and 3.6.
+* Python 3.4 support has been dropped. We now only support Python 2.7, 3.5, 3.6 and 3.7.
+* We will be dropping Python 2.7 support on the next release.
```
[the remark this answers is not in the document]
Comment (wasade) [reply ChangeLog.md]: 💥

File: setup.py (hunk 1 above)
Comment (wasade) [reply setup.py]: Done

File: setup.py (hunk 2 above)
Comment (wasade) [reply setup.py]: Donezo

**Comment by wasade** [comment]
@ElDeveloper, good to merge?

**Pull request merged**